### ISSUE TITLE
glsl: matrix qualifier on shader attributes

### ISSUE THREAD

**Issue by farnoy** [shader-slang/slang#3447]
This slang code currently miscompiles:

```
[shader("vertex")]
float4 main(float3 position, float4x4 mvp) : SV_Position {
    return mul(mvp, float4(position, 1.0)));
}
```

It compiles to something like this GLSL code:

```glsl
#version 450
layout(location = 0)
in vec3 _S1;

layout(row_major) // glslang produces: error: 'layout' : matrix or packing qualifiers can only be used on a uniform or buffer
layout(location = 1)
in mat4x4 _S2;

void main()
{
    gl_Position = (((((vec4(_S1, 1.0)) * (_S2))));
    return;
}
```

**Comment by csyonghe** [comment]
Reading matrices from an input stream is very rare. Do you intend this, or do you mean `uniform float4x4 mvp`?

**Comment by csyonghe** [comment]
GLSL does not allow matrices as vertex stream elements IIRC.

**Comment by csyonghe** [comment]
The fix here is to issue an error message that forbids defining a non-uniform matrix typed parameter.

**Comment by farnoy** [comment]
Yes I did intend to do that and it's useful with per-instance vertex input
attributes. It worked just fine with glsl, I'm pretty sure it's legal.

**Comment by csyonghe** [comment]
The issue here is that the code is being compiled with "column-major" layout, which maps to glsl's "row-major". Can you share you compilation options? Compiling with default row-major layout should get rid of the decoration.

You can also try -emit-spirv-directly and see if it fixes the problem.

**Comment by farnoy** [comment]
My args are `-I src/shaders -g3 -line-directive-mode none -entry main -target glsl -force-glsl-scalar-layout -o out.glsl -- shader.slang`

I'm not passing the `-matrix-layout-row-major` option. Are you saying that if I did pass it, there would be no `layout(row_major)` or `layout(column_major)` annotations on these GLSL attributes? I may be approaching it incorrectly in that case, since I translated my GLSL's `matrix * vector` snippets into `mul(matrix, vector)` in Slang.
